packages/realm-react/src/__tests__/useQueryRender.test.tsx

```diff
@@ -354,4 +371,34 @@ describe.each`
 
     expect(queryByText("756c")).toBeNull();
   });
+  // This replicates the issue https://github.com/realm/realm-js/issues/4375
+  it("will handle multiple async transactions", async () => {
+    const { queryByTestId } = await setupTest({ queryType, useUseObject: true });
+    const asyncEffect = async () => {
+      testRealm.write(() => {
+        testRealm.deleteAll();
+      });
+      let i = 0;
+      while (i < 10) {
+        await new Promise((resolve) => setTimeout(resolve, 10));
+        const id = i;
+        testRealm.write(() => {
+          testRealm.create(Item, { id, name: `${id}` }, Realm.UpdateMode.Modified);
+        });
+        testRealm.write(() => {
+          const item = testRealm.objectForPrimaryKey(Item, id);
+          if (item) {
+            item.name = `${id + 100}`;
+          }
+        });
```

Comment:
So do we only see the issue if we create an item _and_ update it in the same "event loop"? Just curious because the original user's repro was a bit different (created an object, `await sleep(0)`, created another object), if this is the easiest way to repro then all good

Reply:
The `sleep(0)` had the same result, so we could potentially add that back in just to make sure that also is fixed.  It's a bit of a strange concept, since this is already on the event loop.  Curious about how those are tasks get prioritised.  A `sleep(1)` will hide the issue completely.

Reply:
I've added the `sleep(0)` so that this code mirrors the issue explained by the user.